### PR TITLE
Added Grain domain model and entity.

### PR DESCRIPTION
**Pull request by michaelvidal24**

### CODE REVIEW

File: Fabric.Authorization.Domain/Models/Grain.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        
+        public string Name { get; set; }
+
+        public bool IsShared { get; set; }
+
+        public ICollection<SecurableItem> SecurableItems { get; set; }
+
+        public ICollection<string> RequiredWriteScopes { get; set; }
+
+        public string Identifier => Id.ToString();
```
Comment (kpaul) [inline Fabric.Authorization.Domain/Models/Grain.cs]: i thought we were moving away from including the identifier in entities?

File: Fabric.Authorization.Domain/Models/Grain.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Fabric.Authorization.Domain.Models
+{
+    public class Grain : IIdentifiable, ITrackable, ISoftDelete
+    {
+        public Grain()
+        {
```
Comment (bsmithvt) [inline Fabric.Authorization.Domain/Models/Grain.cs]: Should we initialize RequiredWriteScopes here too?

File: Fabric.Authorization.Domain/Models/Grain.cs (hunk 1 above)
Comment (michaelvidal24) [reply Fabric.Authorization.Domain/Models/Grain.cs]: I can remove that, my brain was thinking consistency :-)

File: Fabric.Authorization.Persistence.SqlServer/Fabric.Authorization.Persistence.SqlServer.csproj [hunk 3]
```diff
@@ -26,7 +26,6 @@
   <ItemGroup>
     <Folder Include="Configuration\" />
     <Folder Include="DependencyInjection\" />
```
Comment (bsmithvt) [inline Fabric.Authorization.Persistence.SqlServer/Fabric.Authorization.Persistence.SqlServer.csproj]: Looks like the Mappers folder was removed.

File: Fabric.Authorization.Persistence.SqlServer/Fabric.Authorization.Persistence.SqlServer.csproj [hunk 4]
```diff
@@ -26,7 +26,6 @@
...
-    <Folder Include="Mappers\" />
```
Comment (kpaul) [inline Fabric.Authorization.Persistence.SqlServer/Fabric.Authorization.Persistence.SqlServer.csproj]: was this intentional?

Comment (michaelvidal24) [reply Fabric.Authorization.Persistence.SqlServer/Fabric.Authorization.Persistence.SqlServer.csproj]: No - VS removed it, not sure why. Not all of our folders are marked as include.

**Pull request merged**